hazelcast/src/test/java/com/hazelcast/map/nearcache/NearCacheTest.java

```diff
@@ -196,6 +198,49 @@ public void testNearCacheStats() throws Exception {
     }
 
     @Test
+    public void testHeapCostCalculationWhenConcurrentCacheMisses() {
+        String mapName = randomName();
+        Config config = getConfig();
+        NearCacheConfig nearCacheConfig = newNearCacheConfig().setCacheLocalEntries(true);
+        config.getMapConfig(mapName).setNearCacheConfig(nearCacheConfig.setInvalidateOnChange(false));
+
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(1);
+        HazelcastInstance hazelcastInstance = factory.newHazelcastInstance(config);
+
+        final IMap<Integer, Integer> map = hazelcastInstance.getMap(mapName);
+
+        int threadCount = 10;
+        final int itemCount = 100;
+
+        ExecutorService executorService = Executors.newFixedThreadPool(10);
+        for (int i = 0; i < itemCount; i++) {
+            map.put(i, i);
+        }
+
+        final CountDownLatch countDownLatch = new CountDownLatch(threadCount);
+        Runnable task = new Runnable() {
+            @Override
+            public void run() {
+                for (int i = 0; i < itemCount; i++) {
+                    map.get(i);
+                }
+                countDownLatch.countDown();
+            }
+        };
+
+        for (int i = 0; i < threadCount; i++) {
+            executorService.execute(task);
+        }
+        assertOpenEventually(countDownLatch);
+
+        for (int i = 0; i < itemCount; i++) {
+            map.remove(i);
```

Comment:
Looks like any put/remove on member will invalidate the cache directly. I think otherwise it would be absurd (you could remove an item and get would still return it). InvalidateOnChange is only for changes from other members, maybe the JavaDoc should be updated accordingly?